Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/Merchant.hs

```diff
@@ -1596,6 +1598,8 @@ postMerchantSpecialLocationGatesUpsert _merchantShortId _city specialLocationId
       latitude <- fromMaybeM (InvalidRequest "Latitude cannot be empty for a new gate") $ reqT.latitude <|> (mbGate <&> (.point.lat))
       longitude <- fromMaybeM (InvalidRequest "Longitude cannot be empty for a new gate") $ reqT.longitude <|> (mbGate <&> (.point.lon))
       address <- fromMaybeM (InvalidRequest "Address cannot be empty for a new gate") $ reqT.address <|> (mbGate >>= (.address))
+      mbMerchant <- QM.findByShortId merchantShortId
+      mbMerchantOperatingCity <- CQMOC.findByMerchantShortIdAndCity merchantShortId city
```

Comment:
you can use mid and mocid from specialLoction instead of making these two queries:
line 1579:
`specialLocation <-  QSL.findById specialLocationId >>= fromMaybeM (InvalidRequest "Cound not find a special location with the provided id")`
line 1614-1615:

```
            merchantId = specialLocation.merchantId,
            merchantOperatingCityId = specialLocation.merchantOperatingCityId,
```